### PR TITLE
control-api / redis race condition

### LINKED ISSUE #814: Control API init no longer fails if the script fails
I think maybe due to the new responses from cli, the init job doesn't recognize a failure anymore - I saw the init job appear to succeed but all of the services in the dashboard were red.

In the logs of the init pod every apply had failed: 

```bash
Found service: slo/
Creating mesh object: domain.
Creating object with domain-egress.json
Error: POST request to http://control-api:5555/v1.0/domain returned http status of 500

Object already exists! Editing domain-egress.json
Error: PUT request to http://control-api:5555/v1.0/domain/_ returned http status of 400
----------
Creating object with domain.json
Error: POST request to http://control-api:5555/v1.0/domain returned http status of 500

Object already exists! Editing domain.json
Error: PUT request to http://control-api:5555/v1.0/domain/_ returned http status of 400
```

Upgrading fabric with everything up easily fixed this.

If this script gets updated we should also remove these other harmless errors:

```bash
Adding egress clusters
ls: internal-cluster-*.json: No such file or directory
Adding egress shared_rules
ls: internal-shared-rules-*.json: No such file or directory
Adding additional Special Routes
```

### LINKED ISSUE #841: control-api-init job displays error even though all mesh configs have been applied
While deploy a fresh install of greymatter release-2.3 branch on OCP4.6, control-api-init job completes with the below errors>
The job create a mesh configs successfully but showing these failed requests

```
Configuring mesh from config directory: /etc/config/mesh/
Command Name: greymatter
Version:      2.0.0
Branch:
Commit:       97d56a1
Built:         by
Config dir contains:
edge
services
special
Starting mesh configuration ...
Creating service configuration objects...
Found service: catalog/
Creating mesh object: domain.
Creating object with domain-egress.json
Error: POST request to http://control-api:5555/v1.0/domain returned http status of 500

Object already exists! Editing domain-egress.json
Error: PUT request to http://control-api:5555/v1.0/domain/_ returned http status of 400
----------
Creating object with domain.json
Error: POST request to http://control-api:5555/v1.0/domain returned http status of 500

Object already exists! Editing domain.json
Error: PUT request to http://control-api:5555/v1.0/domain/_ returned http status of 400
----------
Creating mesh object: cluster.
Creating object with cluster.json
Error: POST request to http://control-api:5555/v1.0/cluster returned http status of 400

Object already exists! Editing cluster.json
Error: PUT request to http://control-api:5555/v1.0/cluster/_ returned http status of 400
```

### PR DESCRIPTION
**Pull request by dgoldstein1**
resolves #841 
resolves #814 

**Please provide enough information so that others can review your pull request:**

1. Explain the **details** for making this change. What existing problem does the pull request solve? If it resolves an existing issue, be sure to use a Github [keyword](https://help.github.com/en/articles/closing-issues-using-keywords) to automatically close it.

- remove centos redis image. Currently working in openshift
- add redis health check to control-api-init container

<br/><br/>

2. What **changes to custom.yaml** are required?

<br/><br/>

3. Have you documented any additional setup steps or configurations options?

<br/><br/>

4. Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI.

<br/><br/>

- install in openshift. Change `environment: openshift` and `domain: XXX` to the domain you install to
- install in eks as normal

### CODE REVIEW

File: fabric/control-api/templates/control-api-init-hook.yaml [hunk 1]
```diff
@@ -24,6 +24,16 @@ spec:
               value: {{ .Release.Namespace }}
             - name: SERVICE
               value: {{ .Values.controlApi.name }}
+        {{- if .Values.global.external_redis.disabled }}        
+        - name: ensure-mesh-redis
+          image: {{ .Values.global.waiter.image }}
```
Comment (dontlaugh) [inline fabric/control-api/templates/control-api-init-hook.yaml:29]: Is this reference to "waiter" right?

File: fabric/values.yaml [hunk 2]
```diff
@@ -155,7 +155,11 @@ control:
         cpu: 250m
         memory: 256Mi
 
+
```
Comment (dontlaugh) [inline fabric/values.yaml:158]:
```suggestion
```
Remove this extra line

**Review by dontlaugh — COMMENTED** [review]
Seems alright. I'll want SREs to look.

File: fabric/control-api/templates/control-api-init-hook.yaml (hunk 1 above)
Comment (dgoldstein1) [reply fabric/control-api/templates/control-api-init-hook.yaml:29]: I think so? Which waiter are you referring to?

Comment (dontlaugh) [reply fabric/control-api/templates/control-api-init-hook.yaml:29]: I see now. I don't like that we have to spawn containers to block startup. But we will need to fix that in software.

This isn't a blocker from me to merge this PR :+1:

**Review by dontlaugh — APPROVED** [review]

File: global.yaml [hunk 3]
```diff
@@ -2,11 +2,11 @@ global:
   # Used as imagePullSecrets value for each subchart
   image_pull_secret: docker.secret
   # Deployment environment, one of "eks", "kubernetes", or "openshift"
-  environment: kubernetes
+  environment: openshift
```
Comment (moosecanswim) [inline global.yaml:5]: @dgoldstein1 could you change this back to kubernetes (as a default)

File: global.yaml [hunk 4]
```diff
@@ -2,11 +2,11 @@ global:
   # Used as imagePullSecrets value for each subchart
   image_pull_secret: docker.secret
   # Deployment environment, one of "eks", "kubernetes", or "openshift"
-  environment: kubernetes
+  environment: openshift 
   # Used to configure control and control-api environment variables
   zone: zone-default-zone
   # domain used to template catalog's apiEndpoint cluster config and configure edge
-  domain: localhost:30000
+  domain: greymatter.apps.rp.greymatter.services
```
Comment (moosecanswim) [inline global.yaml:9]: could you also reset this to the default.

**Review by moosecanswim — CHANGES_REQUESTED** [review]

**Comment by dgoldstein1** [comment]
@moosecanswim thanks! I reset `global.yaml`

**Review by moosecanswim — APPROVED** [review]
Works as expected.  Tested on oc4.6 and k3d clusters

**Pull request merged**